solutions/silver/usaco-990.mdx

```diff
@@ -2,20 +2,31 @@
 id: usaco-990
 source: USACO Silver 2020 January
 title: Berry Picking
-author: Qi Wang (Java), Daniel Suh (C++), Juheon Rhee (Python)
+author: Qi Wang, Daniel Suh, Juheon Rhee
 ---
 
+<Spoiler title="Hint 1">
+Let $k$ equal the maximum number of berries in one basket.
+What can we say about the capacity of the rest of the baskets?
```

Comment:
i might just be rusty w/ this problem but
what does max # of berries? isn't it just the largest tree?

Reply:
you don't have to take the entire tree ?

and motivation is that if you have a $k$-basket, then you should try to make as many $k$-baskets as possible.

Reply:
hb uh
"say our baskets were [1,2,3,4,5]. is there any difference between this basket set and [1, 2, 3, 3, 3]?"
not sure if this encourages the mindset for the solution but

Reply:
yeah that works. it does kinda give it away though. maybe we should leave drawing out the cases to the reader?

Reply:
"is it _really_ always the best for bessie to take all the berries from the trees?"

Reply:
ummm wait how does that relate

we want the reader to figure out that given an upper bound for each basket, we want as many baskets to reach this bound. the trees are just an interference in achieving this.

Reply:
how would they come upon the idea for an upper bound though, that seems so jank

Reply:
well if we give the maximum half of the baskets to bessie, then we want to other half to contain as much as possible, which implies that.

Reply:
well, my point is when i read this hint at first it's just like "k is just the largest tree bruh"

Reply:
oh ok, wdyt abt something like:

---

Let's say Bessie and Elsie get the same number of berries, what's a case that give us this? What can we do if this construction doesn't exist?

Reply:
how can a construction suddenly not exist

Reply:
Let's say Bessie and Elsie get the same number of berries, what's a case that give us this? How can we increase this value?

Reply:
[oopsies i forgot we weren't fixing any values]

Reply:
like bessie & elsie both get k berries?

Reply:
yea

Reply:
yeah that seems nice